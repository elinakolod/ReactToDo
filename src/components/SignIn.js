//import axios from 'axios';

//class SignIn extends Component {
  // constructor(props) {
  //   super(props)
  //   this.state = {
  //     headers: []
  //   }
  // }

  // getTasks() {
  //   let params = {
  //     email: 'test10@i.ua',
  //     password: 'BoraBora1',
  //     password_confirmation: 'BoraBora1'
  //   }

  //   axios.post('/auth', params)
  //   .then(response => {
  //     this.setState({headers: response.headers})
  //   })
  //   .catch(error => console.log(error))
  // }

  // componentDidMount() {
  //   this.getTasks()
  // }

  import React, {Component} from 'react';
  import {connect} from 'react-redux';
  import {signInUserRequest} from '../redux/actions';

  class SignIn extends Component {
    state = {
      email: "",
      password: ""
    }

    handleChange = event => {
      this.setState({
        [event.target.name]: event.target.value
      });
    }

    handleSubmit = event => {
      event.preventDefault()
      this.props.signInUserRequest(this.state)
    }

    render() {
      return (
        <form onSubmit={this.handleSubmit}>
          <h1>Login</h1>

          <label>Email</label>
          <input
            name='email'
            placeholder='email'
            value={this.state.email}
            onChange={this.handleChange}
            /><br/>

          <label>Password</label>
          <input
            type='password'
            name='password'
            placeholder='Password'
            value={this.state.password}
            onChange={this.handleChange}
            /><br/>

          <input type='submit'/>
        </form>
      )
    }
  }

  const mapDispatchToProps = {
    signInUserRequest: signInUserRequest
  }

  export default connect(null, mapDispatchToProps)(SignIn);
