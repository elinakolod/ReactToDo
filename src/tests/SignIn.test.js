import React from 'react';
import SignIn from '../components/SignIn'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const middlewares = [thunk];
const mockStore = configureStore(middlewares)
const state = {}
const store = mockStore(state)

it('matches snapshot', () => {
  const wrapper = renderer.create(<Provider store={store}><SignIn/></Provider>)

  expect(wrapper.toJSON()).toMatchSnapshot();
});
