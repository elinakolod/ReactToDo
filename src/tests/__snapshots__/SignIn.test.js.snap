// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <h1>
    Login
  </h1>
  <hr />
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formBasicEmail"
    >
      Email
    </label>
    <input
      className="form-control"
      id="formBasicEmail"
      name="email"
      onChange={[Function]}
      placeholder="Enter email"
      type="email"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formBasicPassword"
    >
      Password
    </label>
    <input
      className="form-control"
      id="formBasicPassword"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Submit
  </button>
</form>
`;
